data/cache/impl.py

```diff
@@ -244,19 +244,32 @@ class RedisDataModelCache(DataModelCache):
     """
 
     def __init__(
-        self,
-        cache_config,
-        host="127.0.0.1",
-        port=6379,
-        password=None,
-        db=0,
-        ca_cert=None,
-        ssl=False,
+            self,
+            cache_config,
+            read_host="127.0.0.1",
+            read_port=6379,
+            write_host="127.0.0.1",
+            write_port=6379,
+            password=None,
+            db=0,
+            ca_cert=None,
+            ssl=False,
```

Comment:
The replica may have different config value for these parameters?

Reply:
I don't think the replica can have a different password or DB. Not sure about CA cert but I'd don't see a use case where we will have a different CA for primary and replica

Reply:
Not sure about passwords, but I'm assuming if a replica can be setup on a separate/completely different host, it should be able to have a separate cert for validation.
We should create a wrapper around the separate configs instead (maybe a namedtuple):
```
_RedisConfig = namedtuple("_RedisConfig", ["host", "port", "password", ...])
```

Reply:
`redis-py` supports single host config only. We would never pass more than 2 hosts to the Redis ModelCache. It's simple if we just pass them via the constructor